### ISSUE TITLE
🐛 Bug Report: Error using Query.search with listFiles

### ISSUE THREAD

**Issue by dj-fiorex** [appwrite/appwrite#8326]
### 👟 Reproduction steps

When i want to search for a particular file, i will use 
```
	const storage = new Storage(client);
	const result = await storage.listFiles(
		bucketId, // bucketId
		[Query.search('name', "some-name")] // queries (optional)
	);
```
This call will fire an error:
`Error: Searching by attribute "name" requires a fulltext index.`

I already go in the database and i think that the fulltext index is already present in _1_bucket_2

<img width="898" alt="image" src="https://github.com/appwrite/appwrite/assets/15034741/efbf5920-b5d7-4f39-ab7b-9909008a8325">

### 👍 Expected behavior

Documentation say that:

Array of query strings generated using the Query class provided by the SDK. Maximum of 100 queries are allowed, each 4096 characters long. **You may filter on the following attributes: name, signature, mimeType, sizeOriginal, chunksTotal, chunksUploaded**

### 👎 Actual Behavior

It throws an error
`Error: Searching by attribute "name" requires a fulltext index.`

### 🎲 Appwrite version

Version 1.5.x

### 💻 Operating system

Linux

### 🧱 Your Environment

VPS OVH cloud
4CPU + 4GB RAM + 80GB NVME
debian 11 + docker

```
_APP_ENV=production
_APP_LOCALE=en
_APP_OPTIONS_ABUSE=enabled
_APP_OPTIONS_FORCE_HTTPS=disabled
_APP_OPTIONS_FUNCTIONS_FORCE_HTTPS=disabled
_APP_OPTIONS_ROUTER_PROTECTION=disabled
_APP_OPENSSL_KEY_V1=<key>
_APP_DOMAIN=<domain>
_APP_DOMAIN_FUNCTIONS=<domain>
_APP_DOMAIN_TARGET=<domain>
_APP_CONSOLE_WHITELIST_ROOT=enabled
_APP_CONSOLE_WHITELIST_EMAILS=<email>
_APP_CONSOLE_WHITELIST_IPS=
_APP_CONSOLE_HOSTNAMES=
_APP_SYSTEM_EMAIL_NAME=Appwrite
_APP_SYSTEM_EMAIL_ADDRESS=<email>
_APP_SYSTEM_RESPONSE_FORMAT=
_APP_SYSTEM_SECURITY_EMAIL_ADDRESS=<email>
_APP_USAGE_STATS=enabled
_APP_LOGGING_PROVIDER=
_APP_LOGGING_CONFIG=
_APP_USAGE_AGGREGATION_INTERVAL=30
_APP_USAGE_TIMESERIES_INTERVAL=30
_APP_USAGE_DATABASE_INTERVAL=900
_APP_WORKER_PER_CORE=6
_APP_REDIS_HOST=redis
_APP_REDIS_PORT=6379
_APP_REDIS_USER=
_APP_REDIS_PASS=
_APP_DB_HOST=mariadb
_APP_DB_PORT=3306
_APP_DB_SCHEMA=appwrite
_APP_DB_USER=user
_APP_DB_PASS=<pass>
_APP_DB_ROOT_PASS=<pass>
_APP_INFLUXDB_HOST=influxdb
_APP_INFLUXDB_PORT=8086
_APP_STATSD_HOST=telegraf
_APP_STATSD_PORT=8125
_APP_SMTP_HOST=
_APP_SMTP_PORT=
_APP_SMTP_SECURE=
_APP_SMTP_USERNAME=
_APP_SMTP_PASSWORD=
_APP_SMS_PROVIDER=
_APP_SMS_FROM=
_APP_STORAGE_LIMIT=30000000
_APP_STORAGE_PREVIEW_LIMIT=20000000
_APP_STORAGE_ANTIVIRUS=disabled
_APP_STORAGE_ANTIVIRUS_HOST=
_APP_STORAGE_ANTIVIRUS_PORT=
_APP_STORAGE_DEVICE=local
_APP_STORAGE_S3_ACCESS_KEY=
_APP_STORAGE_S3_SECRET=
_APP_STORAGE_S3_REGION=us-east-1
_APP_STORAGE_S3_BUCKET=
_APP_STORAGE_DO_SPACES_ACCESS_KEY=
_APP_STORAGE_DO_SPACES_SECRET=
_APP_STORAGE_DO_SPACES_REGION=us-east-1
_APP_STORAGE_DO_SPACES_BUCKET=
_APP_STORAGE_BACKBLAZE_ACCESS_KEY=
_APP_STORAGE_BACKBLAZE_SECRET=
_APP_STORAGE_BACKBLAZE_REGION=us-west-004
_APP_STORAGE_BACKBLAZE_BUCKET=
_APP_STORAGE_LINODE_ACCESS_KEY=
_APP_STORAGE_LINODE_SECRET=
_APP_STORAGE_LINODE_REGION=eu-central-1
_APP_STORAGE_LINODE_BUCKET=
_APP_STORAGE_WASABI_ACCESS_KEY=
_APP_STORAGE_WASABI_SECRET=
_APP_STORAGE_WASABI_REGION=eu-central-1
_APP_STORAGE_WASABI_BUCKET=
_APP_FUNCTIONS_SIZE_LIMIT=30000000
_APP_FUNCTIONS_TIMEOUT=900
_APP_FUNCTIONS_BUILD_TIMEOUT=900
_APP_FUNCTIONS_CONTAINERS=10
_APP_FUNCTIONS_CPUS=0
_APP_FUNCTIONS_MEMORY=0
_APP_FUNCTIONS_MEMORY_SWAP=0
_APP_FUNCTIONS_RUNTIMES=node-20.0
_APP_EXECUTOR_SECRET=<key>
_APP_EXECUTOR_HOST=http://appwrite-executor/v1
_APP_EXECUTOR_RUNTIME_NETWORK=appwrite_runtimes
_APP_FUNCTIONS_ENVS=node-20.0
_APP_FUNCTIONS_INACTIVE_THRESHOLD=60
DOCKERHUB_PULL_USERNAME=
DOCKERHUB_PULL_PASSWORD=
DOCKERHUB_PULL_EMAIL=
OPEN_RUNTIMES_NETWORK=appwrite_runtimes
_APP_FUNCTIONS_RUNTIMES_NETWORK=runtimes
_APP_DOCKER_HUB_USERNAME=
_APP_DOCKER_HUB_PASSWORD=
_APP_FUNCTIONS_MAINTENANCE_INTERVAL=3600
_APP_VCS_GITHUB_APP_NAME=<name>
_APP_VCS_GITHUB_PRIVATE_KEY=<key>
_APP_VCS_GITHUB_APP_ID=<id>
_APP_VCS_GITHUB_CLIENT_ID=<id>
_APP_VCS_GITHUB_CLIENT_SECRET=<key>
_APP_VCS_GITHUB_WEBHOOK_SECRET=<key>
_APP_MAINTENANCE_INTERVAL=86400
_APP_MAINTENANCE_DELAY=0
_APP_MAINTENANCE_RETENTION_CACHE=2592000
_APP_MAINTENANCE_RETENTION_EXECUTION=1209600
_APP_MAINTENANCE_RETENTION_AUDIT=1209600
_APP_MAINTENANCE_RETENTION_ABUSE=86400
_APP_MAINTENANCE_RETENTION_USAGE_HOURLY=8640000
_APP_MAINTENANCE_RETENTION_SCHEDULES=86400
_APP_GRAPHQL_MAX_BATCH_SIZE=10
_APP_GRAPHQL_MAX_COMPLEXITY=250
_APP_GRAPHQL_MAX_DEPTH=3
_APP_MIGRATIONS_FIREBASE_CLIENT_ID=
_APP_MIGRATIONS_FIREBASE_CLIENT_SECRET=
_APP_ASSISTANT_OPENAI_API_KEY=
```

### 👀 Have you spent some time to check if this issue has been raised before?

- [x] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by ItzNotABug** [comment]
Can you instead try doing like this -
```js
const storage = new Storage(client);
const result = await storage.listFiles(
    bucketId, undefined, 'some-name'
);
```

**Comment by dj-fiorex** [comment]
Hello,
i just tried this 
```
  const storage = new Storage(client);
  const result = await storage.listFiles(
    aberlinkBucketId, // bucketId
    [],
    `WC0369` // search (optional)
  );
```
in my bucket i have 3 files with the following names:

- Jun_carcasse_WC0369_22_18.06-1.xlsx
- Jun_carcasse_WC0369_22_18.06.xlsx
- Jun_carcasse_WB0349_22_18.06.xlsx

But the query didn't return anything!

What's wrong guys? i need this feature to be able to ship my app to the client 

Thanks

**Comment by Souptik2001** [comment]
Hey @dj-fiorex,

Let me try to break it down for you.

TLDR; Read the last point for the solution.

- Firslty answering the question, why did the code you provided in the issue description didn't work -
```js
const storage = new Storage(client);
const result = await storage.listFiles(
    bucketId, // bucketId
    [Query.search('name', "some-name")] // queries (optional)
);
```
I am unsure how you got that `name` as a FULLTEXT index in your table structure. But in reality `name` column is not indexed as full-text.
If you see the indices of a file collection, then you will see none of the index contains name attribute - https://github.com/appwrite/appwrite/blob/ae87033974a700b4c2cd6e8935a436bae0d61670/app/config/collections.php#L5817-L5874

Therefore you are getting the error that the attribute name requires a full-text index.

- Now talking about, why the search solution didn't work for your use case -
```js
  const storage = new Storage(client);
  const result = await storage.listFiles(
    aberlinkBucketId, // bucketId
    [],
    `WC0369` // search (optional)
  );
```

So, for that you have to understand that how the search value is internally used for full-text search. So, if you know for full-text search `MATCH()` and `AGAINST()`  is used.
The WHERE clause for this one is created something like this -
```SQL
MATCH(<attribute>) AGAINST('WC0369*' IN BOOLEAN MODE)
```

As you can see the wild-card is only added to the end, therefore it can only match for something like - `WC0369<something_here>` and not `<something_here>WC0369`. This is a limitation of SQL's full-text search only.
You can see the utopia-php database library (which Apprwrite maintains and uses) to know more.

That's the reason this also didn't work as you want to search for file with name `Jun_carcasse_WC0369_22_18.06-1.xlsx`.

- Here comes CONTAINS to your rescue -
  For your usecase try this -
  ```JS
  const storage = new Storage(client);
  const result = await storage.listFiles(
    aberlinkBucketId, // bucketId
    [new Query('contains', 'name', 'WC0369')]
  );
  ```
It will basically to a wild-card query like this - `%WC0369%` on the name column/attribute. Which will return result which has this part of the string in its name.

**But please note that this will not be as performant as full-text search. Full-text search is extreme performance, which you will not get from this.**


FYI - @stnguyen90 @ItzNotABug Please correct me if I am wrong at any point.
